homeassistant/components/zwave_js/discovery.py

```diff
@@ -14,51 +14,143 @@
+    endpoint: Optional[Set[int]] = None
+    # [optional] the value's property must match ANY of these values
+    property: Optional[Set[Union[str, int]]] = None
+    # [optional] the value's metadata_type must match ANY of these values
+    type: Optional[Set[str]] = None
+
+
 @dataclass
 class ZWaveDiscoverySchema:
     """Z-Wave discovery schema.
 
-    The (primary) value for an entity must match these conditions.
+    The Z-Wave node and it's (primary) value for an entity must match these conditions.
     Use the Z-Wave specifications to find out the values for these parameters:
     https://github.com/zwave-js/node-zwave-js/tree/master/specs
     """
 
     # specify the hass platform for which this scheme applies (e.g. light, sensor)
     platform: str
+    # primary value belonging to this discovery scheme
+    primary_value: ZWaveValueDiscoverySchema
     # [optional] hint for platform
     hint: Optional[str] = None
+    # [optional] the node's manufacturer_id must match ANY of these values
+    manufacturer_id: Optional[Set[int]] = None
+    # [optional] the node's product_id must match ANY of these values
+    product_id: Optional[Set[int]] = None
+    # [optional] the node's product_type must match ANY of these values
+    product_type: Optional[Set[int]] = None
+    # [optional] the node's firmware_version must match ANY of these values
+    firmware_version: Optional[Set[str]] = None
     # [optional] the node's basic device class must match ANY of these values
     device_class_basic: Optional[Set[str]] = None
     # [optional] the node's generic device class must match ANY of these values
     device_class_generic: Optional[Set[str]] = None
     # [optional] the node's specific device class must match ANY of these values
     device_class_specific: Optional[Set[str]] = None
-    # [optional] the value's command class must match ANY of these values
-    command_class: Optional[Set[int]] = None
-    # [optional] the value's endpoint must match ANY of these values
-    endpoint: Optional[Set[int]] = None
-    # [optional] the value's property must match ANY of these values
-    property: Optional[Set[Union[str, int]]] = None
-    # [optional] the value's metadata_type must match ANY of these values
-    type: Optional[Set[str]] = None
+    # [optional] additional values that need to be present on the node for this scheme to pass
+    required_values: Optional[Set[ZWaveValueDiscoverySchema]] = None
+    # [optional] bool to specify if this primary value may be discovered by multiple platforms
+    allow_multi: bool = False
 
 
 # For device class mapping see:
 # https://github.com/zwave-js/node-zwave-js/blob/master/packages/config/config/deviceClasses.json
 DISCOVERY_SCHEMAS = [
+    # ====== START OF DEVICE SPECIFIC MAPPING SCHEMAS =======
+    # Honeywell/Jasco 39358 In-Wall Fan Control using switch multilevel CC
+    ZWaveDiscoverySchema(
+        platform="fan",
+        manufacturer_id={57},
```

Comment:
Can we make these (4-digit?) hex instead? Basically everywhere that shows these writes them in hex, the ZWave Alliance DB, Open-Zwave and zwave-js. Easier to reference later.

```suggestion
        manufacturer_id={0x0039},
```

Reply:
Yeah, no problem. I think we don't even need to convert to int (we receive int from the server).
Can you test with the hex value if discovery still works ?